io/io/src/TROSharedMemFile.cxx

```diff
@@ -0,0 +1,53 @@
+// @(#)root/io:$Id$
+// Author: Axel Naumann, 2018-03-17
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+/**
+\class TROSharedMemFile TROSharedMemFile.cxx
+\ingroup IO
+
+A TROSharedMemFile is like a normal TMemFile except that it reads from shared memory.
+*/
+
+#include "TROSharedMemFile.h"
+#include "TError.h"
+
+////////////////////////////////////////////////////////////////////////////////
+/// Usual Constructor.  See the TFile constructor for details.
+
+TROSharedMemFile::TROSharedMemFile(const char *name, std::shared_ptr<const std::vector<char>> data) :
+   TMemFile(name, const_cast<char*>(data->data()), data->size(), "READ", "read-only memfile"),
```

Comment:
doesn't this copy the buffer inside `TMemFile` (which is what we want to avoid)?